### ISSUE TITLE
Full support for Flux attention masking

### ISSUE THREAD

**Issue by squewel** [huggingface/diffusers#10194]
**Is your feature request related to a problem? Please describe.**
SimpleTuner/Kohya allow T5 attention masked training, however this is not currently supported natively in diffusers

**Describe the solution you'd like.**
Already implemented and used in Simpletuner and Kohya: https://github.com/bghira/SimpleTuner/blob/main/helpers/models/flux/transformer.py

**Describe alternatives you've considered.**
Recent implementation doesn't really solve the use case of using existing fine tunes with attention masking with diffusers 
https://github.com/huggingface/diffusers/pull/10122

**Additional context.**
@yiyixuxu @bghira @AmericanPresidentJimmyCarter 

@bghira's suggestion: "i'd suggested they add encoder_attention_mask and image_attention_mask and if image_attention_mask is None that they could then 1-fill those positions and just cat them together"

**Comment by bghira** [comment]
i guess this could live concurrently to the joint_kwargs implementation and just check_inputs error on whether the encoder or image attn mask are passed in alongside it.

**Comment by dxqbYD** [comment]
I'm not sure the extent of this problem is realized. This is not "only" about finetuners having to use their own code to support attention masking.

It completely breaks prompt adherence for the Flux diffusers pipeline.

You can simulate the effects of attention masking (or the lack thereof) by setting a shorter `max_sequence_length`:

```
import torch
from diffusers import FluxPipeline

pipe = FluxPipeline.from_pretrained(
    "black-forest-labs/FLUX.1-dev",
    torch_dtype=torch.bfloat16
).to('cuda')

prompt = "Portrait photo of an angry man"
prompt_T5_token_length = 7

generator = torch.Generator(device='cuda').manual_seed(0)
images = pipe(
    prompt=prompt,
    guidance_scale=3.5,
    height=1024,
    width=1024,
    num_inference_steps=20,
    generator=generator,
).images

images[0].save(f"attention-to-padding.jpg")

generator = torch.Generator(device='cuda').manual_seed(0)
images = pipe(
    prompt=prompt,
    guidance_scale=3.5,
    height=1024,
    width=1024,
    num_inference_steps=20,
    generator=generator,
    max_sequence_length = prompt_T5_token_length,
).images

images[0].save(f"no-attention-to-padding.jpg")
```

attention-to-padding.jpg, most of the attention is to padding:
![Image](https://github.com/user-attachments/assets/d0fd15ae-a284-4886-9288-a60b78bc87fa)

no-attention-to-padding.jpg, actually following the prompt:
![Image](https://github.com/user-attachments/assets/0568bc88-a1a0-4cfe-839c-64bf0f6503a3)

This is exacerbated by the fact that the diffusers pipeline always seems to use the full max_sequence_length, even on short prompts with batch size 1 - but with proper attention masking, this would only be inefficient, not change the outcome.
